alternator/expressions.cc

```diff
@@ -418,9 +418,14 @@ void for_condition_expression_on(const parsed::condition_expression& ce, const n
 // calculate_size() is ConditionExpression's size() function, i.e., it takes
 // a JSON-encoded value and returns its "size" as defined differently for the
 // different types - also as a JSON-encoded number.
-// It return a JSON-encoded "null" value if this value's type has no size
-// defined. Comparisons against this non-numeric value will later fail.
-static rjson::value calculate_size(const rjson::value& v) {
+// If the value's type (e.g. number) has no size defined, there are two cases:
+// 1. If from_data (the value came directly from an attribute of the data),
+//    It returns a JSON-encoded "null" value. Comparisons against this
+//    non-numeric value will later fail, so eventually the application will
+//    get a ConditionalCheckFailedException.
+// 2. Otherwise (the value came from a constant in the query or some other
+//    calculation), throw a ValidationException.
+static rjson::value calculate_size(const rjson::value& v, bool from_data) {
```

Comment:
to me from_data name doesn't tell me anything, I'd just call it is_path because I know what path is in alternator, and data is very broad term, but it's matter of taste, one can probably find counterarguments easily

Reply:
The name "from_data" was supposed to suggest that the value `v` came "from the data", i.e., from what was stored in the database, in contrast to "from the query" (a constant or function call). You're right that from_data really means that v came from a path (i.e., attribute). It's not that it **is** a path, though, but that it came from a path.

I'll consider how to rename it. Maybe "from_path", or even "v_from_path" to make it even clearer that v came from a path, and not something else?